signature/src/bls_over_bn254.rs

```diff
@@ -746,4 +817,14 @@ mod tests {
             Result::Err(SignatureError::FailedValidityCheck),
         );
     }
+
+    #[test]
+    fn test_constant_time_scalar_mul_g1() {
```

Comment:
other than correctness, can you also try to add a constant-timeness test? run multiple times, the drift shouldnt' be noticeable. (like within 0.000001 sec away or sth)